### PR TITLE
Make some lints incremental.

### PR DESCRIPTION
**Pull request by cjgillot**
Those lints do not track a state, so don't need to be performed for the full crate at once.

### CODE REVIEW

**Comment by oli-obk** [comment]
That's some nice wins

@bors r+

File: compiler/rustc_middle/src/mir/interpret/queries.rs [hunk 1]
```diff
@@ -171,6 +171,39 @@ impl<'tcx> TyCtxtAt<'tcx> {
     }
 }
 
+impl<'tcx> TyCtxtEnsure<'tcx> {
+    /// Evaluates a constant without providing any substitutions. This is useful to evaluate consts
+    /// that can't take any generic arguments like statics, const items or enum discriminants. If a
+    /// generic parameter is used within the constant `ErrorHandled::ToGeneric` will be returned.
```
Comment (RalfJung) [inline compiler/rustc_middle/src/mir/interpret/queries.rs:177]: These comments (also the one below) are wrong now -- they talk about the return value of a function that doesn't return anything! I have no idea what is going on here so if someone could make a PR to update the comments, that would be much appreciated. :)

**Pull request merged**